cpp/src/bindings.cpp

```diff
@@ -147,36 +146,28 @@ PYBIND11_MODULE(_pypowsybl, m) {
             .export_values();
 
     py::class_<load_flow_parameters, std::shared_ptr<load_flow_parameters>>(m, "LoadFlowParameters")
-            .def(py::init([](pypowsybl::VoltageInitMode voltageInitMode, bool transformerVoltageControlOn, bool noGeneratorReactiveLimits,
-                             bool phaseShifterRegulationOn, bool twtSplitShuntAdmittance, bool simulShunt, bool readSlackBus, bool writeSlackBus,
-                             bool distributedSlack, pypowsybl::BalanceType balanceType, bool dcUseTransformerRatio, const std::vector<std::string>& countriesToBalance,
-                             pypowsybl::ConnectedComponentMode connectedComponentMode) {
-                auto parameters = new load_flow_parameters();
-                parameters->voltage_init_mode = voltageInitMode;
-                parameters->transformer_voltage_control_on = transformerVoltageControlOn;
-                parameters->no_generator_reactive_limits = noGeneratorReactiveLimits;
-                parameters->phase_shifter_regulation_on = phaseShifterRegulationOn;
-                parameters->twt_split_shunt_admittance = twtSplitShuntAdmittance;
-                parameters->simul_shunt = simulShunt;
-                parameters->read_slack_bus = readSlackBus;
-                parameters->write_slack_bus = writeSlackBus;
-                parameters->distributed_slack = distributedSlack;
-                parameters->balance_type = balanceType;
-                parameters->dc_use_transformer_ratio = dcUseTransformerRatio;
-                parameters->countries_to_balance = pypowsybl::copyVectorStringToCharPtrPtr(countriesToBalance);
-                parameters->countries_to_balance_count = countriesToBalance.size();
-                parameters->connected_component_mode = connectedComponentMode;
+            .def(py::init([]() {
+                 auto parameters = new load_flow_parameters();
+                 load_flow_parameters* config_params = pypowsybl::readParameters();
```

Comment:
There will be a small memory leak here!
you'll need to delete this at the end (or you can embed it in a smart pointer so that it does the job for you :) ).

Other option : maybe the `readParameters` could be an `updateParameters` instead which would take an existing `parameters` as input and update it on the java side ?